clash-lib/src/Clash/Unique.hs

```diff
@@ -421,6 +422,13 @@ subsetUniqSet
   -> Bool
 subsetUniqSet (UniqSet e1) (UniqSet e2) = IntMap.null (IntMap.difference e1 e2)
 
+-- | Check whether A and B are disjoint
+disjointUniqSet
+  :: UniqSet a
+  -> UniqSet a
+  -> Bool
+disjointUniqSet (UniqSet e1) (UniqSet e2) = IntMap.disjoint e1 e2
```

Comment:
TODO: It seems `IntMap.disjoint` doesn't exist in the version of `containers` used by GHC 8.6. The definition + relevant copyright notice should be added to a `Data.IntMap.Strict.Extra` module with a note saying it can be removed when GHC 8.6 is no longer supported